src/de.rs

```diff
@@ -704,22 +720,25 @@ where
             0xbf => self.parse_indefinite_map(visitor),
 
             // Major type 6: optional semantic tagging of other major types
-            0xc0..=0xd7 => self.recursion_checked(|de| de.parse_value(visitor)),
+            0xc0..=0xd7 => {
+                let tag = byte as usize - 0xc0;
```

Comment:
Instead: `let tag = byte as u64 - 0xc0;` (Avoid converting from u8 to usize to u64)